projects/commudle-admin/src/app/feature-modules/users/components/public-profile/user-basic-details/basic-user-profile/social-links/social-links.component.ts

```diff
@@ -0,0 +1,58 @@
+import { Component, EventEmitter, OnInit, Output } from '@angular/core';
+import { FormBuilder } from '@angular/forms';
+import { ICurrentUser } from 'projects/shared-models/current_user.model';
+import { LibAuthwatchService } from 'projects/shared-services/lib-authwatch.service';
+import { UserProfileManagerService } from 'projects/commudle-admin/src/app/feature-modules/users/services/user-profile-manager.service';
+import { faBehance, faDribbble, faFacebook, faGitlab, faMediumM, faYoutube } from '@fortawesome/free-brands-svg-icons';
+
+@Component({
+  selector: 'app-social-links',
+  templateUrl: './social-links.component.html',
+  styleUrls: ['./social-links.component.scss'],
+})
+export class SocialLinksComponent implements OnInit {
+  currentUser: ICurrentUser;
+  @Output() socialLinksFormValidity: EventEmitter<boolean> = new EventEmitter<boolean>();
+
+  socialLinksForm = this.fb.group({
```

Comment:
Let's add some regex validations for the links

Reply:
we are using a library which takes the username and convert it into a valid url automatically